diff --git a/Makefile b/Makefile
index 8cacaaf7..52bf76b6 100644
--- a/Makefile
+++ b/Makefile
@@ -62,6 +62,8 @@ else ifneq (,$(findstring odroid,$(platform)))
    override platform += unix
 else ifneq (,$(findstring webos,$(platform)))
    override platform += unix
+else ifneq (,$(findstring @DEVICE@,$(platform)))
+   override platform += unix
 endif
 
 # system platform
@@ -169,6 +171,20 @@ ifneq (,$(findstring unix,$(platform)))
 
    endif
    
+   #@DEVICE@
+   ifneq (,$(findstring @DEVICE@,$(platform)))
+     ifeq (,$(findstring AMD64,$(platform)))
+       TARGET := $(TARGET_NAME)_libretro.so
+       CPUFLAGS += -DARM_FIX -DCLASSIC -DARM64
+       WITH_DYNAREC = aarch64
+     endif
+   endif
+   #AMD64
+   ifneq (,$(findstring AMD64,$(platform)))
+     TARGET := $(TARGET_NAME)_libretro.so
+     WITH_DYNAREC = x86
+   endif
+
    # ODROIDs
    ifneq (,$(findstring odroid,$(platform)))
       BOARD ?= $(shell cat /proc/cpuinfo | grep -i odroid | awk '{print $$3}')
